src/python/tests/appengine/handlers/external_update_test.py

```diff
@@ -0,0 +1,211 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for external_update."""
+
+import base64
+import json
+import os
+import unittest
+
+import flask
+import webtest
+
+from datastore import data_types
+from handlers import external_update
+from tests.test_libs import helpers as test_helpers
+from tests.test_libs import test_utils
+
+DATA_DIRECTORY = os.path.join(os.path.dirname(__file__), 'external_update_data')
+
+
+@test_utils.with_cloud_emulators('datastore')
+class ExternalUpdatesTest(unittest.TestCase):
+  """Test external updates."""
+
+  def _read_test_data(self, name):
+    """Helper function to read test data."""
+    with open(os.path.join(DATA_DIRECTORY, name), 'rb') as handle:
+      return handle.read()
+
+  def setUp(self):
+    flaskapp = flask.Flask('testflask')
+    flaskapp.add_url_rule(
+        '/external-update', view_func=external_update.Handler.as_view(''))
+    self.app = webtest.TestApp(flaskapp)
+
+    test_helpers.patch(self, [
+        'google.oauth2.id_token.verify_oauth2_token',
+    ])
+
+    self.mock.verify_oauth2_token.return_value = {
+        'email_verified': True,
+        'email': 'test-clusterfuzz@appspot.gserviceaccount.com',
+    }
+
+    data_types.Job(
+        name='external_job',
+        external_reproduction_topic='topic',
+        external_updates_subscription='subscription').put()
+    data_types.Job(name='job').put()
+
+    self.testcase_0 = data_types.Testcase(
```

Comment:
nit: just testcase instead of testcase_0

Reply:
Fixed!